src/App/Page.php

```diff
@@ -407,13 +408,16 @@ public function exit(ResponseInterface $response)
 	 * @param Mode                        $mode     The current node mode
 	 * @param ResponseInterface           $response The Response of the module class, including type, content & headers
 	 * @param L10n                        $l10n     The l10n language class
+	 * @param Profiler                    $profiler
 	 * @param IManageConfigValues         $config   The Configuration of this node
 	 * @param IManagePersonalConfigValues $pconfig  The personal/user configuration
-	 * @param int                         $localUID The UID of the local user
-	 *
-	 * @throws HTTPException\InternalServerErrorException|HTTPException\ServiceUnavailableException
+	 * @param IHandleUserSessions         $session
+	 * @param Nav                         $nav
+	 * @throws HTTPException\MethodNotAllowedException
+	 * @throws HTTPException\InternalServerErrorException
+	 * @throws HTTPException\ServiceUnavailableException
 	 */
-	public function run(App $app, BaseURL $baseURL, Arguments $args, Mode $mode, ResponseInterface $response, L10n $l10n, Profiler $profiler, IManageConfigValues $config, IManagePersonalConfigValues $pconfig, int $localUID)
+	public function run(App $app, BaseURL $baseURL, Arguments $args, Mode $mode, ResponseInterface $response, L10n $l10n, Profiler $profiler, IManageConfigValues $config, IManagePersonalConfigValues $pconfig, IHandleUserSessions $session, Nav $nav)
```

Comment:
Why did you replace the int with the interface? I thought it would be better to pass rather basic variables than complex if possible (reducing complexity)

Reply:
In one of the iterations of this branch, I was calling the static `Nav::build` method with the full session object as parameter. When I went further and made the whole `Nav` class dynamic, its instantiation moved elsewhere but I didn't think about simplifying back this specific parameter since it was still working as expected.